logical/system_view.go

```diff
@@ -37,6 +39,17 @@ type SystemView interface {
 
 	// ReplicationState indicates the state of cluster replication
 	ReplicationState() consts.ReplicationState
+
+	// ResponseWrapData wraps the given data in a cubbyhole and returns the
+	// token used to unwrap.
+	ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (string, error)
```

Comment:
I think this function should take in not just data to put into the cubbyhole but also any claims data you want to put into the token itself (signed plaintext data).